### PR TITLE
ISPN-15111 mapping of HS5 annotations to Infinispan native

### PR DESCRIPTION
**Pull request by domiborges**
Add information about mapping HS5 annotations to Infinispan native annotations

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/upgrading.adoc [hunk 1]
```diff
@@ -220,14 +220,34 @@ If Hibernate 5.3 support is needed, you must use an older version of Infinispan.
...
+| @Field(store = Store.YES) | projectable = true | Fields previously marked as stored are now projectable.
+
+| type String && @Field(analyze = Analyze.YES) | analyzer = "<definition>"| String fields that were marked with analyzer definitions remain being analyzed during indexing.
+
+| @Field(analyze = Analyze.NO) && (@Field(store = Store.YES) OR @Field(sortable = Sortable.YES)) | sortable = true| Fields that were not analyzed but were either stored in the index or marked as sortable are now considered sortable.
+
+| N/A | aggregable = true| Performing aggregation operations using the Hibernate 5 Search annotations was not possible.
+
+| N/A | normalizer = "lowercase"| Mapping fields that were analyzed or normalized is not possible due to the potential data loss in the process.
+|===
```
Comment (domiborges) [inline documentation/src/main/asciidoc/topics/upgrading.adoc:250]: The table looks as follows
image

File: documentation/src/main/asciidoc/topics/upgrading.adoc [hunk 2]
```diff
@@ -220,14 +220,34 @@ If Hibernate 5.3 support is needed, you must use an older version of Infinispan.
...
+| HS5 annotations | {brandname} native annotations| Description
+
+| @Field(index=Index.YES) | searchable | Fields previously marked as indexed are now searchable.
+
+| @Field(store = Store.YES) | projectable = true | Fields previously marked as stored are now projectable.
+
+| type String && @Field(analyze = Analyze.YES) | analyzer = "<definition>"| String fields that were marked with analyzer definitions remain being analyzed during indexing.
+
+| @Field(analyze = Analyze.NO) && (@Field(store = Store.YES) OR @Field(sortable = Sortable.YES)) | sortable = true| Fields that were not analyzed but were either stored in the index or marked as sortable are now considered sortable.
```
Comment (domiborges) [inline documentation/src/main/asciidoc/topics/upgrading.adoc:245]: I hope I can replace "||" with "OR" It's necessary because of the asciidoc syntax

**Comment by domiborges** [comment]
Please consider this as a draft @fax4ever and let's see how we can further improve it.

**Review by fax4ever — APPROVED** [review]
LGTM. Thank you @domiborges.
I have just a suggestion here.

File: documentation/src/main/asciidoc/topics/upgrading.adoc (hunk 2 above)
Comment (fax4ever) [reply documentation/src/main/asciidoc/topics/upgrading.adoc:245]: I think it is OK

File: documentation/src/main/asciidoc/topics/upgrading.adoc (hunk 1 above)
Comment (fax4ever) [reply documentation/src/main/asciidoc/topics/upgrading.adoc:250]: Maybe we can have a column with the native annotations in which are possible to define the feature. E.g.: for  the`analyzer="<definition>"` row we can add `@Text`.

**Comment by domiborges** [comment]
I added annotations in the table @fax4ever 
Are there any manual steps users need to do?

**Comment by fax4ever** [comment]
> I added annotations in the table @fax4ever Are there any manual steps users need to do?

The change looks good to me.
No they are not.
Going to merge it

**Comment by domiborges** [comment]
Thank you @fax4ever

**Comment by fax4ever** [comment]
Thank you for the contribution!

**Pull request merged**